SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPCTypes.h

```diff
@@ -138,19 +156,70 @@ class RPCBufferReceiver
 public:
 	virtual ~RPCBufferReceiver() = default;
 
-	virtual void OnAdded(Worker_EntityId EntityId, EntityViewElement const& Element);
-	virtual void OnAdded_ReadComponent(const RPCReadingContext& iCtx) = 0;
+	virtual void OnAdded(FName ReceiverName, Worker_EntityId EntityId, EntityViewElement const& Element);
+	virtual void OnAdded_ReadComponent(const RPCReadingContext& Ctx) = 0;
 	virtual void OnRemoved(Worker_EntityId EntityId) = 0;
 	virtual void OnUpdate(const RPCReadingContext& iCtx) = 0;
-	virtual void FlushUpdates(RPCWritingContext&) = 0;
-	virtual void ExtractReceivedRPCs(const RPCCallbacks::CanExtractRPCs&, const RPCCallbacks::ProcessRPC&) = 0;
+	virtual void FlushUpdates(RPCWritingContext& Ctx) = 0;
 
 	const TSet<Worker_ComponentId>& GetComponentsToRead() const { return ComponentsToRead; }
 
 protected:
 	TSet<Worker_ComponentId> ComponentsToRead;
 };
 
+struct RPCEmptyData
+{
+};
+
+template <typename T>
+struct NullReceiveWrapper
+{
+	using AdditionalData = RPCEmptyData;
+	struct WrappedData
+	{
+		WrappedData(T&& InData)
+			: Data(MoveTemp(InData))
+		{
+		}
+
+		const AdditionalData& GetAdditionalData() const
+		{
+			static EmptyData s_Dummy;
+			return s_Dummy;
+		}
+
+		const T& GetData() const { return Data; }
+
+		T Data;
+	};
+
+	WrappedData MakeWrappedData(Worker_EntityId EntityId, T&& Data, uint64 RPCId) { return MoveTemp(Data); }
+};
+
+template <typename T, template <typename> class PayloadWrapper = NullReceiveWrapper>
```

Comment:
Looks like PayloadWrapper always specializes with T, could this be just PayloadWrapper = NullReceiveWrapper<T>?

Reply:
It could be, but I find it nicer to use a template wrapper, instead of having to write the payload type name twice.

TQueue<Payload, Wrapper> vs TQueue<Payload, Wrapper<Payload>>

Reply:
Oh interesting, it got destroyed by formatting but I get what you mean - not seen this pattern before but it's neat